File: 1078 - Occurrence After Bigram.py


class Solution:
    def findOcurrences(self, text: str, first: str, second: str) -> list[str]:
        text = text.split()
        res = []
        a, b = '', ''
        for word in text:
            if a == first and b == second:
                res.append(word)
            a = b
            b = word
        return res
        

File: 1078 - Occurrence After Bigram.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public String[] findOcurrences(String text, String first, String second) {
        String l1 = "", l2 = "";
        List<String> res = new ArrayList<>();
        String[] words = text.split(" ");
        
        for (String word: words) {
            if (l1.equals(first) && l2.equals(second)) {
                res.add(word);
            }
            l1 = l2;
            l2 = word;
        }
        return res.toArray(new String[0]);
    }
}